### ISSUE TITLE
Bug beim Durchstellen

### ISSUE THREAD

**Issue by merlink01** [sipgate/sipgate.io#75]
Wenn ein Kunde auf einen Kollegen (A) Anruft und angenommen wird, 
wird als user dieser Kollege (A) in onAnswer übermittelt.

Stellt diese Kollege diesen Kunden nun an einen an einen anderen Kollegen (B) durch,
werden diese Informationen momentan gar nicht übermittelt.

Hier wäre das Verhalten, das ich erwarten würde:

-User A hat den Kunden Angenommen: Verhalten wie bisher
-User A Klingelt bei Kollege B an: Ein eigenständiger Anruf von Kollege A zu B wird Generiert (ringing + onAnswer)
-User A legt auf und übergiebt: Beide Aktiven Gespräche werden beendet und ein neues Gespräch (B zu Kunden wird signalisiert)

**Comment by maxnowack** [comment]
+1

**Comment by dennisoderwald** [comment]
+1

**Comment by danielberlin** [comment]
Ein neues Event für's Durchstellen wäre allerdings ebenso denkbar und zumindest auf den Anruf "Kunde-User A" sauberer, da dieser Anruf dann direkt übergeben und nicht beendet und als neuer Anruf neu aufgebaut wird.

Der Anruf 1 "Kunde-A" wird dann anders als oben skizziert nicht beendet (sondern nur der Anruf 2 "A-B"), sondern für Anruf 1 wird event=transfer übermittelt. Hierbei könnte man ggf. noch beide CallIds übermitteln.

**Comment by JHGitty** [comment]
+1

**Comment by findingmarbles** [comment]
Haben wir auffem Schirm. Wollen wir langfristig machen, hat aber niedrige Prio. Daher schliessen wir das Ticket erstmal.

**Comment by merlink01** [comment]
Ich würde die Priorität nicht zu niedrig ansetzen.
Man sollte bedenken, dass jegliche Statistiken (z.B. wie viel telefoniert jeder Mitarbeiter) komplett verfälscht werden.

Des weiteren Benötigen wir diese Informationen um festzustellen, welcher Mitarbeiter frei zum weiterleiten ist.

Auch diese Informationen stimmen nach dem durchstellen nicht mehr.

Dies sind schon grundlegende Funktionen einer professionellen Telefonanlage und und keine unwichtigen Features.

**Comment by maxnowack** [comment]
Ich stimme @merlink01 zu. Bitte die niedrige Priorität nochmal überdenken!

**Comment by dennisoderwald** [comment]
+1 @maxnowack - dem ist nichts mehr hinzuzufügen, außer der Sprachlosigkeit der Einstufung. Bitte überdenken!

**Comment by danielberlin** [comment]
+1

**Comment by HashDot** [comment]
+1

**Comment by tobiashieb** [comment]
+1 

Das w#re auch für unseren Usecase absolut elementar.

**Comment by findingmarbles** [comment]
Danke für eure Kommentare!
Wir haben nochmal drüber geredet und wollen es kurz- oder mittelfristig umsetzen :)

**Comment by merlink01** [comment]
Perfekt, vieln Dank.

**Comment by maxnowack** [comment]
Gibt's schon was neues? :)

**Comment by tobiashieb** [comment]
Interessante Frage. Dann könnten wir mit der Integration loslegen, liebe Corinna. :)

**Comment by findingmarbles** [comment]
Sorry, wir sind momentan an einer anderen Baustelle. Nach der, soll es dann mit dieser Issue weitergehen. Posten wir dann natürlich hier.

**Comment by merlink01** [comment]
Wie sieht es denn mit diesem Bug aus?
Nun ist ja doch schon ein halbes jahr rum und Mitarbeiter die einen Anruf durchgestellt haben werden immer noch als belegt dargestellt.

**Comment by danielberlin** [comment]
Wir wäre es, bei einem Anruf, der von Kollege A angenommen wird und von diesem an Kollege B (mit Rückfrage) durchgestellt wird, ein `event=newCall` nur zu Beginn und dann ein `event=transfer` zu übermitteln, dieses jedoch mit derselben callId zu versehen? Und dann entsprechende `event=answer` und `hangup` wie gehabt. Fände ich die sauberste Lösung.

**Comment by merlink01** [comment]
OK, es hat sich was verändert. 
Die Partei, die durgestellt und aufgelegt hat ist nun markiert als cancelled.
Das sieht schon mal gut aus :-)

Habt ihr API changes vorgenommen?

**Comment by wiesson** [comment]
@merlink01 Ja, wir sind gerade dabei. Weitere Informationen folgen spätestens morgen!

**Comment by CookieCrumb23** [comment]
Wie von @wiesson angekündigt, haben wir hierfür etwas implementiert. Wie sich sipgate.io beim Durchstellen verhält ist jetzt in der Readme nachzulesen.
https://github.com/sipgate/sipgate.io/blob/master/README.md#faq

**Comment by CookieCrumb23** [comment]
@danielberlin,

kannst du mir ein Beispiel für so einen Anruf, bei dem das `from=anonymous` an daniel@sipgate.io schicken?

Und was meinst du mit `callId` ist original? In meinen Versuchen das Verhalten nach zu stellen habe ich für beide newCall Events unterschiedliche callIds erhalten.

**Comment by danielberlin** [comment]
Ich meinte, dass ich o.g. `callId` nicht anonymisiert habe, d.h. unter dieser findest Du exakt diesen Beispielanruf bereits im System.

**Comment by CookieCrumb23** [comment]
@danielberlin 

Ah, verstehe. Da wir das sipgate.io Log nur 24 Stunden lang vorhalten und ich den Anruf im sipgate.io Log nicht finde (ich gehe davon aus, dass der älter als 24 Stunden ist?), benötige ich leider das Datum , die Uhrzeit für den Anruf und die Kundennummer des Accounts um das Ganze nachvollziehen zu können.

**Comment by merlink01** [comment]
Meiner Meinung nach ist das hier Fixed. :-)

**Comment by danielberlin** [comment]
Der von mir hier am 19. Juni beschriebene Bug besteht fort. Gerade letzte Woche gesehen.

**Comment by merlink01** [comment]
Ok, reopened.

**Comment by merlink01** [comment]
Ok, reopened.

**Comment by olivermg** [comment]
> Der unter README.md#faq im ersten Beispiel unter Punkt 3 beschriebene `newCall` enthält `from=anonymous` statt der Rufnummer des Weiterleitenden:
> 
> `event=newCall&direction=in&from=anonymous&to=123&callId=9191897610291272706`

Dieses Problem besteht weiterhin.
